### PR TITLE
Changed the AdminClientProvider to use keystore and password in theSecrets

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Refactoring

### Description

Each Secret containing certificates and keys has now the corresponding store as well (with related password).
This PR refactors the `AdminClientProvider` and the default implementation in order to leverage them instead of recreating needed stores from certificate and key and generating a new password.
It also allows specifying which store "key" to be used in order to allow not only the CO using it (with its cluster operator Secret) but even UO (with entity operator Secret).

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by ppatierno** [comment]
@strimzi-ci run tests

File: operator-common/src/main/java/io/strimzi/operator/common/AdminClientProvider.java [hunk 1]
```diff
@@ -2,11 +2,11 @@
  * Copyright Strimzi authors.
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
-package io.strimzi.operator.cluster.operator.resource;
+package io.strimzi.operator.common;
 
 import io.fabric8.kubernetes.api.model.Secret;
 import org.apache.kafka.clients.admin.AdminClient;
 
 public interface AdminClientProvider {
```
Comment (tombentley) [inline operator-common/src/main/java/io/strimzi/operator/common/AdminClientProvider.java]: Can we javadoc it, now that it's going to be used in more places?

File: operator-common/src/main/java/io/strimzi/operator/common/AdminClientProvider.java [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
-    AdminClient createAdminClient(String hostname, Secret clusterCaCertSecret, Secret coKeySecret);
+    AdminClient createAdminClient(String hostname, Secret clusterCaCertSecret, Secret keyCertSecret, String keyCertName);
```
Comment (tombentley) [inline operator-common/src/main/java/io/strimzi/operator/common/AdminClientProvider.java]: KIP-476 added a superinterface called `Admin` for the `AdminClient` class, so I think we should make this interface return an `Admin`.

**Comment by ppatierno** [comment]
@strimzi-ci run tests

**Comment by ppatierno** [comment]
@strimzi-ci run tests

**Pull request merged**